src/DrupalFinder.php

```diff
@@ -7,294 +7,49 @@
 
 namespace DrupalFinder;
 
+use Composer\InstalledVersions;
+
 class DrupalFinder
 {
-    /**
-     * Drupal root environment variable.
-     */
-    const ENV_DRUPAL_ROOT = 'DRUPAL_FINDER_DRUPAL_ROOT';
-
-    /**
-     * Composer root environment variable.
-     */
-    const ENV_COMPOSER_ROOT = 'DRUPAL_FINDER_COMPOSER_ROOT';
-
-    /**
-     * Vendor directory environment variable.
-     */
-    const ENV_VENDOR_DIR = 'DRUPAL_FINDER_VENDOR_DIR';
-
-    /**
-     * Drupal web public directory.
-     *
-     * @var string
-     */
-    private $drupalRoot;
-
-    /**
-     * Drupal package composer directory.
-     *
-     * @var bool
-     */
-    private $composerRoot;
-
-    /**
-     * Composer vendor directory.
-     *
-     * @var string
-     *
-     * @see https://getcomposer.org/doc/06-config.md#vendor-dir
-     */
-    private $vendorDir;
-
-    /**
-     * Initialize finder.
-     *
-     * Optionally pass the starting path.
-     *
-     * @param string|null $start_path
-     *   The path to begin the search from.
-     *
-     * @throws \Exception
-     * @todo Make $start_path mandatory in v2.
-     */
-    public function __construct($start_path = null) {
-        // Initialize path variables to false, indicating their locations are
-        // not yet known.
-        $this->drupalRoot = false;
-        $this->composerRoot = false;
-        $this->vendorDir = false;
-
-        // If a starting path was provided, attempt to locate and set path
-        // variables.
-        if (!empty($start_path)) {
-            $this->discoverRoots($start_path);
-        }
-    }
-
-    /**
-     * Locate Drupal, Composer, and vendor directory paths.
-     *
-     * @param string $start_path
-     *   The path to begin the search from.
-     *
-     * @return bool
-     *   True if the Drupal root was identified, false otherwise.
-     *
-     * @throws \Exception
-     *
-     * @deprecated Will be removed in v2. Future usage should instantiate
-     *   a new DrupalFinder object by passing the starting path to its
-     *   constructor.
-     */
-    public function locateRoot($start_path)
-    {
-        $this->discoverRoots($start_path);
-        return !empty($this->getDrupalRoot());
-    }
-
     /**
      * Get the Drupal root.
      *
-     * @return string|bool
      *   The path to the Drupal root, if it was discovered. False otherwise.
      */
-    public function getDrupalRoot()
+    public function getDrupalRoot(): string|bool
```

Comment:
Since this already requires a major version bump, can we use `NULL` to indicate we couldn't find anything? Then this can simply be `?string` which is the more common PHP way.

Reply:
Fixed, and others